src/Evolution/Systems/NewtonianEuler/Sources/LaneEmdenGravitationalField.hpp

```diff
@@ -0,0 +1,91 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"  // IWYU pragma: keep
+#include "Evolution/Systems/NewtonianEuler/TagsDeclarations.hpp"  // IWYU pragma: keep
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+class DataVector;
+
+namespace NewtonianEuler {
+namespace Solutions {
+struct LaneEmdenStar;
+}  // namespace Solutions
+}  // namespace NewtonianEuler
+
+namespace PUP {
+class er;
+}  // namespace PUP
+namespace Tags {
+template <size_t Dim, typename Frame>
+struct Coordinates;
+template <typename SolutionType>
+struct AnalyticSolution;
+}  // namespace Tags
+
+namespace gsl {
+template <typename T>
+class not_null;
+}  // namespace gsl
+/// \endcond
+
+// IWYU pragma: no_include "Evolution/Systems/NewtonianEuler/Tags.hpp"
+
+// IWYU pragma: no_forward_declare Tensor
+
+namespace NewtonianEuler {
+namespace Sources {
+
+/*!
+ * \brief Source giving the acceleration due to gravity in the spherical,
+ * Newtonian Lane-Emden star solution.
+ *
+ * The gravitational field \f$g^i\f$ enters the NewtonianEuler system as a
+ * source term for the conserved momentum,
+ *
+ * \f{align*}
+ * \partial_t S^i + \partial_j F^{j}(S^i) = S(S^i) = \rho g^i,
+ * \f}
+ *
+ * where \f$S^i\f$ is the conserved momentum density, \f$F^{j}(S^i)\f$ is the
+ * flux of \f$S^i\f$, \f$\rho\f$ is the fluid mass density.
+ *
+ * \note This source is specialized to the LaneEmdenStar analytic solution in
+ * that it queries a LaneEmdenStar object for the gravitational field. In all
+ * other respects, the source term is generic.
```

Comment:
Regarding this last point, maybe all these sources derived from an external acceleration profile should be somehow handled generically. I have an idea of how to do this but for now I'd suggest to move on with what is done here.

What is important fact to note here, though, is that this source will *not* integrate the density (the one passed as an argument in the apply function) in order to obtain a self-consistent gravitational field (i.e. as if one were solving Euler + Poisson) but rather will return the source generated by a _given_ acceleration field profile (i.e. a function of x), which, in this case, is computed from a Lane-Emden solution. I'd rephrase this note clearly stating all this.

Reply:
I'll add a note to hopefully make this distinction clearer. I agree it would be nice to bundle the different acceleration-type sources together eventually 🙂